backend/operations_scanner.go

```diff
@@ -0,0 +1,346 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+	"time"
+
+	ocmsdk "github.com/openshift-online/ocm-sdk-go"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	ocmerrors "github.com/openshift-online/ocm-sdk-go/errors"
+
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/ARO-HCP/internal/database"
+	"github.com/Azure/ARO-HCP/internal/ocm"
+)
+
+const (
+	defaultCosmosOperationsPollInterval = 30 * time.Second
+	defaultClusterServicePollInterval   = 10 * time.Second
+)
+
+type OperationsScanner struct {
+	dbClient         database.DBClient
+	lockClient       *database.LockClient
+	clusterService   ocm.ClusterServiceClient
+	activeOperations []*database.OperationDocument
+	done             chan struct{}
+}
+
+func NewOperationsScanner(dbClient database.DBClient, ocmConnection *ocmsdk.Connection) *OperationsScanner {
+	return &OperationsScanner{
+		dbClient:         dbClient,
+		lockClient:       dbClient.GetLockClient(),
+		clusterService:   ocm.ClusterServiceClient{Conn: ocmConnection},
+		activeOperations: make([]*database.OperationDocument, 0),
+		done:             make(chan struct{}),
+	}
+}
+
+func getInterval(envName string, defaultVal time.Duration, logger *slog.Logger) time.Duration {
+	if intervalString, ok := os.LookupEnv(envName); ok {
+		interval, err := time.ParseDuration(intervalString)
+		if err == nil {
+			return interval
+		} else {
+			logger.Warn(fmt.Sprintf("Cannot use %s: %s", envName, err.Error()))
+		}
+	}
+	return defaultVal
+}
+
+func (s *OperationsScanner) Run(ctx context.Context, logger *slog.Logger, stop <-chan struct{}) {
+	defer close(s.done)
+
+	var interval time.Duration
+
+	interval = getInterval("COSMOS_OPERATIONS_POLL_INTERVAL", defaultCosmosOperationsPollInterval, logger)
+	logger.Info("Polling Cosmos Operations items every " + interval.String())
+	pollDBOperationsTicker := time.NewTicker(interval)
+
+	interval = getInterval("CLUSTER_SERVICE_POLL_INTERVAL", defaultClusterServicePollInterval, logger)
+	logger.Info("Polling Cluster Service every " + interval.String())
+	pollCSOperationsTicker := time.NewTicker(interval)
+
+	// Poll database immediately on startup.
+	s.pollDBOperations(ctx, logger)
+
+	for {
+		select {
+		case <-pollDBOperationsTicker.C:
+			s.pollDBOperations(ctx, logger)
+		case <-pollCSOperationsTicker.C:
+			s.pollCSOperations(ctx, logger, stop)
+		case <-stop:
+			break
+		}
+	}
+}
+
+func (s *OperationsScanner) Join() {
+	<-s.done
+}
+
+func (s *OperationsScanner) pollDBOperations(ctx context.Context, logger *slog.Logger) {
+	activeOperations := make([]*database.OperationDocument, 0)
```

Comment:
Not sure if there is a specific reason for this declaration, but the idiomatic way is `var activeOperations []*database.OperationDocument` - it more clearly indicates that a zero-length slice is being allocated. The `make` path is correct but adds review load to the PR as reviewers need to read carefully to see what the intent is.